resources/example-msg-templates/slack_default.j2

```diff
@@ -0,0 +1,226 @@
+{# You can set any mandatory tags here, and they will be formatted/outputted in the message #}
+{% set requiredTags = ['Application','Owner'] %}
+{% set expireDays = 90 %}
+
+{# The macros below format some resource attributes for better presentation #}
+{%- macro getTag(resource, tagKey) -%}
+  {%- if resource.get('Tags') -%}
+    {%- for t in resource.get('Tags') -%}
+      {%-  if t.get('Key') == tagKey -%}
+        {{ t.get('Value') }}
+      {%- endif -%}
+    {%- endfor -%}
+  {%- endif -%}
+{%- endmacro -%}
+
+{% macro slackData(resources, columnNames) -%}
+  {%- for resource in resources %}
+    {%- for columnName in columnNames -%}
+    {% set inner_loop = loop %}
+      {%- if columnName in requiredTags -%}
+        {%- set value = getTag(resource,columnName) -%}
+      {%- elif columnName == 'tag.Name' -%}
+        {%- set value = getTag(resource,'Name') -%}
+      {%- elif columnName == 'InstanceCount' -%}
+        {%- set value = resource['Instances'] | length -%}
+      {%- elif columnName == 'AccessKeys:RemainDays' -%}
+        {%- set combinedkeys = dict() -%}
+        {%- for accesskeydet in resource['c7n:AccessKeys'] -%}
+          {%- set accesskeyid = accesskeydet['AccessKeyId'] -%}
+          {%- set remaindays = ( expireDays - (accesskeydet['CreateDate'] | get_date_age)) -%}
+          {%- set _dummy = combinedkeys.update({ accesskeyid:remaindays }) -%}
+        {%- endfor -%}
+        {%- set value = combinedkeys -%}
+      {%- elif columnName == 'RoleLastUsed' -%}
+        {%- if resource['RoleLastUsed'].get('LastUsedDate') -%}
+          {%- set value = resource['RoleLastUsed'].get('LastUsedDate')  + ' (Last used ' ~ (resource['RoleLastUsed'].get('LastUsedDate')  | get_date_age)  + ' days ago)' -%}
+        {%- else -%}
+          {%- set value = "Never used -" + ' (Created ' ~ (resource['CreateDate'] | get_date_age)  + ' days ago)' -%}
+        {%- endif -%}
+      {%- elif columnName == 'PublicIp' -%}
+        {%- set value = resource['NetworkInterfaces'][0].get('Association')['PublicIp'] -%}
+      {%- elif columnName == 'PublicIpAddress' -%}
+        {%- set value = resource['PublicIp'] -%}
+      {%- else -%}
+        {% set value = resource[columnName] %}
+      {%- endif -%}
+      {%- if  inner_loop.index == 1 -%}
+        {{"\n"}}*- {{ columnName }}*: {{ value }}
+      {%- else %}
+    {{"\n     "}}*{{ columnName }}*: {{ value }}
+      {%- endif -%}
+    {%- endfor -%}
+  {%- endfor -%}
+{% endmacro %}
+
+{# Main #}
+{%- macro createData(columnNames, resources) -%}
```

Comment:
nit: love the idea of having a consistent top-level macro to format resource data, though I wonder if another name (`formatResourceData`, `formatResources`, etc?) would convey its purpose more clearly. For context, the original default HTML template uses `createTable` - I like how yours is more flexible, but we may be able to find a middle ground with the naming :).